<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Matt Bradley" />

<meta name="date" content="2024-11-20" />

<title>Detecting Copy Number Variation on Targeted Exon Sequencing with RCNA</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Detecting Copy Number Variation on Targeted
Exon Sequencing with RCNA</h1>
<h4 class="author">Matt Bradley</h4>
<h4 class="date">2024-11-20</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>RCNA was designed with the purpose of detecting copy number
alterations on targeted exon sequencing. While detecting copy number
alterations are typically difficult when using discrete support, the
RCNA package uses a conservative neighbor-based approach to provide
robust results where previous low cost methods such as RNA microarrays
cannot. While the regions of interest are not specifically required to
be genes (or even exons for that matter) the example we provide in this
vignette is performed on artificial data modeled after the original
use-case in our paper.</p>
<p>The workflow for RCNA can be summarized as the following: - Estimate
the GC-content-related bias present in the raw coverage data - Correct
the GC-content bias based on the previous estimation - Estimate the
normal karyotype ranges for each gene in the annotation file - Estimate
the copy number alteration ratio based on the normal karyotype range
estimations</p>
<p>These steps can be executed as their own commmands, or using the
<code>run_RCNA</code> function, as seen in the proceeding sections.</p>
</div>
<div id="example-data" class="section level2">
<h2>Example Data</h2>
<p>The data we use for this vignette can be found in the
<code>data/</code> folder under <code>vignette</code>. This dataset has
a number of genes specified in the annotation file, and the sample
names, raw coverage files, and chrX normalization are specified in
<code>sample.csv</code>. Below you can find how we transform the data
from the raw flat files into a special S4 object used for analysis.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(RCNA)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;vignette&quot;</span>, <span class="st">&quot;samples.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;RCNA&quot;</span>), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>new.obj <span class="ot">&lt;-</span> <span class="fu">create_RCNA_object</span>(<span class="at">sample.names =</span> samples<span class="sc">$</span>sample.name, </span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>                        <span class="at">ano.file =</span> <span class="fu">system.file</span>(<span class="st">&quot;vignette&quot;</span> ,<span class="st">&quot;annotations.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;RCNA&quot;</span>),</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>                        <span class="at">out.dir =</span> <span class="st">&#39;output&#39;</span>,</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>                        <span class="at">file.raw.coverage =</span> <span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>, samples<span class="sc">$</span>raw.cov.file),</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>                        <span class="at">norm.cov.matrix =</span> <span class="fu">file.path</span>(<span class="st">&quot;output&quot;</span>, <span class="st">&quot;norm-cov-matrix.csv.gz&quot;</span>),</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>                        <span class="at">nkr =</span> <span class="fl">0.9</span>,</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>                        <span class="at">x.norm =</span> samples<span class="sc">$</span>x.norm,</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>                        <span class="at">low.score.cutoff =</span> <span class="sc">-</span><span class="fl">0.5</span>,</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>                        <span class="at">high.score.cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>                        <span class="at">ncpu =</span> ncpu)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>new.obj</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;RCNA_object&quot;</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co">#&gt; Slot &quot;sample.names&quot;:</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;sample-0001&quot; &quot;sample-0002&quot; &quot;sample-0003&quot; &quot;sample-0004&quot; &quot;sample-0005&quot; &quot;sample-0006&quot; &quot;sample-0007&quot; &quot;sample-0008&quot; &quot;sample-0009&quot; &quot;sample-0010&quot;</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="co">#&gt; Slot &quot;ano.file&quot;:</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/home/mbradley/R/x86_64-pc-linux-gnu-library/4.2/RCNA/vignette/annotations.csv&quot;</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="co">#&gt; Slot &quot;out.dir&quot;:</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="co">#&gt; [1] &quot;output&quot;</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="co">#&gt; Slot &quot;ncpu&quot;:</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="co">#&gt; [1] 4</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a><span class="co">#&gt; Slot &quot;win.size&quot;:</span></span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="co">#&gt; [1] 75</span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a><span class="co">#&gt; Slot &quot;gc.step&quot;:</span></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a><span class="co">#&gt; [1] 0.01</span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="co">#&gt; Slot &quot;estimate_gc&quot;:</span></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a><span class="co">#&gt; Slot &quot;nkr&quot;:</span></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a><span class="co">#&gt; [1] 0.9</span></span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a><span class="co">#&gt; Slot &quot;norm.cov.matrix&quot;:</span></span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a><span class="co">#&gt; [1] &quot;output/norm-cov-matrix.csv.gz&quot;</span></span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a><span class="co">#&gt; Slot &quot;low.score.cutoff&quot;:</span></span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a><span class="co">#&gt; [1] -0.5</span></span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a><span class="co">#&gt; Slot &quot;high.score.cutoff&quot;:</span></span>
<span id="cb1-47"><a href="#cb1-47" tabindex="-1"></a><span class="co">#&gt; [1] 0.5</span></span>
<span id="cb1-48"><a href="#cb1-48" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-49"><a href="#cb1-49" tabindex="-1"></a><span class="co">#&gt; Slot &quot;gcParams&quot;:</span></span>
<span id="cb1-50"><a href="#cb1-50" tabindex="-1"></a><span class="co">#&gt;          file.raw.coverage sample.names                file.corrected.coverage</span></span>
<span id="cb1-51"><a href="#cb1-51" tabindex="-1"></a><span class="co">#&gt; 1  data/sample-0001.txt.gz  sample-0001 output/gc/sample-0001.corrected.txt.gz</span></span>
<span id="cb1-52"><a href="#cb1-52" tabindex="-1"></a><span class="co">#&gt; 2  data/sample-0002.txt.gz  sample-0002 output/gc/sample-0002.corrected.txt.gz</span></span>
<span id="cb1-53"><a href="#cb1-53" tabindex="-1"></a><span class="co">#&gt; 3  data/sample-0003.txt.gz  sample-0003 output/gc/sample-0003.corrected.txt.gz</span></span>
<span id="cb1-54"><a href="#cb1-54" tabindex="-1"></a><span class="co">#&gt; 4  data/sample-0004.txt.gz  sample-0004 output/gc/sample-0004.corrected.txt.gz</span></span>
<span id="cb1-55"><a href="#cb1-55" tabindex="-1"></a><span class="co">#&gt; 5  data/sample-0005.txt.gz  sample-0005 output/gc/sample-0005.corrected.txt.gz</span></span>
<span id="cb1-56"><a href="#cb1-56" tabindex="-1"></a><span class="co">#&gt; 6  data/sample-0006.txt.gz  sample-0006 output/gc/sample-0006.corrected.txt.gz</span></span>
<span id="cb1-57"><a href="#cb1-57" tabindex="-1"></a><span class="co">#&gt; 7  data/sample-0007.txt.gz  sample-0007 output/gc/sample-0007.corrected.txt.gz</span></span>
<span id="cb1-58"><a href="#cb1-58" tabindex="-1"></a><span class="co">#&gt; 8  data/sample-0008.txt.gz  sample-0008 output/gc/sample-0008.corrected.txt.gz</span></span>
<span id="cb1-59"><a href="#cb1-59" tabindex="-1"></a><span class="co">#&gt; 9  data/sample-0009.txt.gz  sample-0009 output/gc/sample-0009.corrected.txt.gz</span></span>
<span id="cb1-60"><a href="#cb1-60" tabindex="-1"></a><span class="co">#&gt; 10 data/sample-0010.txt.gz  sample-0010 output/gc/sample-0010.corrected.txt.gz</span></span>
<span id="cb1-61"><a href="#cb1-61" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-62"><a href="#cb1-62" tabindex="-1"></a><span class="co">#&gt; Slot &quot;nkrParams&quot;:</span></span>
<span id="cb1-63"><a href="#cb1-63" tabindex="-1"></a><span class="co">#&gt;                         file.nkr.coverage x.norm sample.names</span></span>
<span id="cb1-64"><a href="#cb1-64" tabindex="-1"></a><span class="co">#&gt; 1  output/gc/sample-0001.corrected.txt.gz  FALSE  sample-0001</span></span>
<span id="cb1-65"><a href="#cb1-65" tabindex="-1"></a><span class="co">#&gt; 2  output/gc/sample-0002.corrected.txt.gz  FALSE  sample-0002</span></span>
<span id="cb1-66"><a href="#cb1-66" tabindex="-1"></a><span class="co">#&gt; 3  output/gc/sample-0003.corrected.txt.gz  FALSE  sample-0003</span></span>
<span id="cb1-67"><a href="#cb1-67" tabindex="-1"></a><span class="co">#&gt; 4  output/gc/sample-0004.corrected.txt.gz  FALSE  sample-0004</span></span>
<span id="cb1-68"><a href="#cb1-68" tabindex="-1"></a><span class="co">#&gt; 5  output/gc/sample-0005.corrected.txt.gz  FALSE  sample-0005</span></span>
<span id="cb1-69"><a href="#cb1-69" tabindex="-1"></a><span class="co">#&gt; 6  output/gc/sample-0006.corrected.txt.gz  FALSE  sample-0006</span></span>
<span id="cb1-70"><a href="#cb1-70" tabindex="-1"></a><span class="co">#&gt; 7  output/gc/sample-0007.corrected.txt.gz  FALSE  sample-0007</span></span>
<span id="cb1-71"><a href="#cb1-71" tabindex="-1"></a><span class="co">#&gt; 8  output/gc/sample-0008.corrected.txt.gz  FALSE  sample-0008</span></span>
<span id="cb1-72"><a href="#cb1-72" tabindex="-1"></a><span class="co">#&gt; 9  output/gc/sample-0009.corrected.txt.gz  FALSE  sample-0009</span></span>
<span id="cb1-73"><a href="#cb1-73" tabindex="-1"></a><span class="co">#&gt; 10 output/gc/sample-0010.corrected.txt.gz  FALSE  sample-0010</span></span>
<span id="cb1-74"><a href="#cb1-74" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-75"><a href="#cb1-75" tabindex="-1"></a><span class="co">#&gt; Slot &quot;scoreParams&quot;:</span></span>
<span id="cb1-76"><a href="#cb1-76" tabindex="-1"></a><span class="co">#&gt;                       file.score.coverage sample.names</span></span>
<span id="cb1-77"><a href="#cb1-77" tabindex="-1"></a><span class="co">#&gt; 1  output/gc/sample-0001.corrected.txt.gz  sample-0001</span></span>
<span id="cb1-78"><a href="#cb1-78" tabindex="-1"></a><span class="co">#&gt; 2  output/gc/sample-0002.corrected.txt.gz  sample-0002</span></span>
<span id="cb1-79"><a href="#cb1-79" tabindex="-1"></a><span class="co">#&gt; 3  output/gc/sample-0003.corrected.txt.gz  sample-0003</span></span>
<span id="cb1-80"><a href="#cb1-80" tabindex="-1"></a><span class="co">#&gt; 4  output/gc/sample-0004.corrected.txt.gz  sample-0004</span></span>
<span id="cb1-81"><a href="#cb1-81" tabindex="-1"></a><span class="co">#&gt; 5  output/gc/sample-0005.corrected.txt.gz  sample-0005</span></span>
<span id="cb1-82"><a href="#cb1-82" tabindex="-1"></a><span class="co">#&gt; 6  output/gc/sample-0006.corrected.txt.gz  sample-0006</span></span>
<span id="cb1-83"><a href="#cb1-83" tabindex="-1"></a><span class="co">#&gt; 7  output/gc/sample-0007.corrected.txt.gz  sample-0007</span></span>
<span id="cb1-84"><a href="#cb1-84" tabindex="-1"></a><span class="co">#&gt; 8  output/gc/sample-0008.corrected.txt.gz  sample-0008</span></span>
<span id="cb1-85"><a href="#cb1-85" tabindex="-1"></a><span class="co">#&gt; 9  output/gc/sample-0009.corrected.txt.gz  sample-0009</span></span>
<span id="cb1-86"><a href="#cb1-86" tabindex="-1"></a><span class="co">#&gt; 10 output/gc/sample-0010.corrected.txt.gz  sample-0010</span></span>
<span id="cb1-87"><a href="#cb1-87" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-88"><a href="#cb1-88" tabindex="-1"></a><span class="co">#&gt; Slot &quot;commands&quot;:</span></span>
<span id="cb1-89"><a href="#cb1-89" tabindex="-1"></a><span class="co">#&gt; list()</span></span></code></pre></div>
<p><strong>Note:</strong> While the samples and annotation files have
been included with this package, the raw coverage files are not included
in this package as they are larger than what is allowed by CRAN. A
release of the raw data may be included in future updates to this
vignette.</p>
<p>The form of the coverage file should be a space-separated file with
no headers and match the exact format as seen below. The columns should
represent (1) chromosome, (2) position, (3) target ID - this is optional
but not used, (4) coverage in read depth, and (5) the DNA nucleotide
base. This vignette uses 10 samples, each spanning 96 genes and exactly
336,583 base pairs.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">read.table</span>(<span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;sample-0001.txt.gz&quot;</span>), <span class="at">sep =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co">#&gt;     V1      V2   V3   V4 V5</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt; 1 chr1 1787331 r456  977  T</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; 2 chr1 1787332 r456 1004  T</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 3 chr1 1787333 r456  991  A</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; 4 chr1 1787334 r456  994  G</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; 5 chr1 1787335 r456 1049  T</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; 6 chr1 1787336 r456 1022  T</span></span></code></pre></div>
<p>The annotation file must be a comma-separated file with the columns
seen below, including headers. They are (1) the name of the genetic
feature, such as a gene name (2) chromosome, (3) feature start position,
and (4) feature end position. This vignette uses two genes as
features.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;vignette&quot;</span> ,<span class="st">&quot;annotations.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;RCNA&quot;</span>)))</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt;   feature chromosome    start      end</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt; 1   ALAS2          X 55009055 55031064</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; 2   SRSF2         17 76734115 76737374</span></span></code></pre></div>
</div>
<div id="estimate-and-correct-gc-bias" class="section level2">
<h2>Estimate and Correct GC Bias</h2>
<p>The first step to the RCNA workflow is to estimate the GC-content
bias in the raw coverage data. This is done using a sliding window
approach, which first estimates the GC-content. The GC-content is then
normalized against the mean coverage across all bases in the coverage
file. Lastly, a corrective multiplicative factor is calculated for each
GC-content “bin”. Each bin represents a certain percentage of GC-content
in a given window. This correction is then applied iteratively using a
second sliding window. The size of the sliding window (in bp) can be
specified using the <code>win.size</code> parameter (default = 75 bp).
For this example we will use both the default <code>gc.step</code> and
<code>win.size</code> parameters. This step can be accomplished by using
the <code>correct_gc_bias</code> function, as shown below.</p>
<p><strong>Note on performance:</strong> This function will train and
correct the <strong>entire</strong> coverage file. This is beneficial to
the accuracy of the correction, as running the correction on only a few
genes can result in biased results. Keeping this in mind, the run-time
of the algorithm scales linearly in respect to both number of bases as
well as the number of samples causing this step to occupy a vast
majority of the entire workflow’s run-time. We recommend using a
pre-determined GC factor file or providing pre-corrected coverage files
if possible to cut down on the execution time.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>gc.res <span class="ot">&lt;-</span> <span class="fu">correct_gc_bias</span>(new.obj, <span class="at">estimate_gc =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#&gt; No output file column labeled `file.gc.factor` - defaulting to sample names. See `?correct_gc_bias()` for more information.</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; Beginning GC estimation</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt; Calculating GC bias correction</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; Feature score estimation succeeded!</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>gc.res[[<span class="dv">1</span>]]</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;RCNA_analysis&quot;</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; Slot &quot;call&quot;:</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;estimate_gc_bias&quot;</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt; Slot &quot;params&quot;:</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt; $file.raw.coverage</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;data/sample-0001.txt.gz&quot; &quot;data/sample-0002.txt.gz&quot; &quot;data/sample-0003.txt.gz&quot; &quot;data/sample-0004.txt.gz&quot; &quot;data/sample-0005.txt.gz&quot; &quot;data/sample-0006.txt.gz&quot; &quot;data/sample-0007.txt.gz&quot; &quot;data/sample-0008.txt.gz&quot; &quot;data/sample-0009.txt.gz&quot;</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt; [10] &quot;data/sample-0010.txt.gz&quot;</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; $file.gc.factor</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;output/gc/sample-0001.gc.factor.txt&quot; &quot;output/gc/sample-0002.gc.factor.txt&quot; &quot;output/gc/sample-0003.gc.factor.txt&quot; &quot;output/gc/sample-0004.gc.factor.txt&quot; &quot;output/gc/sample-0005.gc.factor.txt&quot; &quot;output/gc/sample-0006.gc.factor.txt&quot;</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;output/gc/sample-0007.gc.factor.txt&quot; &quot;output/gc/sample-0008.gc.factor.txt&quot; &quot;output/gc/sample-0009.gc.factor.txt&quot; &quot;output/gc/sample-0010.gc.factor.txt&quot;</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co">#&gt; Slot &quot;res.files&quot;:</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co">#&gt; $file.gc.factor</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;output/gc/sample-0001.gc.factor.txt&quot; &quot;output/gc/sample-0002.gc.factor.txt&quot; &quot;output/gc/sample-0003.gc.factor.txt&quot; &quot;output/gc/sample-0004.gc.factor.txt&quot; &quot;output/gc/sample-0005.gc.factor.txt&quot; &quot;output/gc/sample-0006.gc.factor.txt&quot;</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;output/gc/sample-0007.gc.factor.txt&quot; &quot;output/gc/sample-0008.gc.factor.txt&quot; &quot;output/gc/sample-0009.gc.factor.txt&quot; &quot;output/gc/sample-0010.gc.factor.txt&quot;</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>gc.res[[<span class="dv">2</span>]]</span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;RCNA_analysis&quot;</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a><span class="co">#&gt; Slot &quot;call&quot;:</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a><span class="co">#&gt; [1] &quot;correct_gc_bias&quot;</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a><span class="co">#&gt; Slot &quot;params&quot;:</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a><span class="co">#&gt; $win.size</span></span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a><span class="co">#&gt; [1] 75</span></span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a><span class="co">#&gt; $gcParams</span></span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a><span class="co">#&gt;          file.raw.coverage                file.corrected.coverage                      file.gc.factor sample.names</span></span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a><span class="co">#&gt; 1  data/sample-0001.txt.gz output/gc/sample-0001.corrected.txt.gz output/gc/sample-0001.gc.factor.txt  sample-0001</span></span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a><span class="co">#&gt; 2  data/sample-0002.txt.gz output/gc/sample-0002.corrected.txt.gz output/gc/sample-0002.gc.factor.txt  sample-0002</span></span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a><span class="co">#&gt; 3  data/sample-0003.txt.gz output/gc/sample-0003.corrected.txt.gz output/gc/sample-0003.gc.factor.txt  sample-0003</span></span>
<span id="cb4-39"><a href="#cb4-39" tabindex="-1"></a><span class="co">#&gt; 4  data/sample-0004.txt.gz output/gc/sample-0004.corrected.txt.gz output/gc/sample-0004.gc.factor.txt  sample-0004</span></span>
<span id="cb4-40"><a href="#cb4-40" tabindex="-1"></a><span class="co">#&gt; 5  data/sample-0005.txt.gz output/gc/sample-0005.corrected.txt.gz output/gc/sample-0005.gc.factor.txt  sample-0005</span></span>
<span id="cb4-41"><a href="#cb4-41" tabindex="-1"></a><span class="co">#&gt; 6  data/sample-0006.txt.gz output/gc/sample-0006.corrected.txt.gz output/gc/sample-0006.gc.factor.txt  sample-0006</span></span>
<span id="cb4-42"><a href="#cb4-42" tabindex="-1"></a><span class="co">#&gt; 7  data/sample-0007.txt.gz output/gc/sample-0007.corrected.txt.gz output/gc/sample-0007.gc.factor.txt  sample-0007</span></span>
<span id="cb4-43"><a href="#cb4-43" tabindex="-1"></a><span class="co">#&gt; 8  data/sample-0008.txt.gz output/gc/sample-0008.corrected.txt.gz output/gc/sample-0008.gc.factor.txt  sample-0008</span></span>
<span id="cb4-44"><a href="#cb4-44" tabindex="-1"></a><span class="co">#&gt; 9  data/sample-0009.txt.gz output/gc/sample-0009.corrected.txt.gz output/gc/sample-0009.gc.factor.txt  sample-0009</span></span>
<span id="cb4-45"><a href="#cb4-45" tabindex="-1"></a><span class="co">#&gt; 10 data/sample-0010.txt.gz output/gc/sample-0010.corrected.txt.gz output/gc/sample-0010.gc.factor.txt  sample-0010</span></span>
<span id="cb4-46"><a href="#cb4-46" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-47"><a href="#cb4-47" tabindex="-1"></a><span class="co">#&gt; $gc.step</span></span>
<span id="cb4-48"><a href="#cb4-48" tabindex="-1"></a><span class="co">#&gt; [1] 0.01</span></span>
<span id="cb4-49"><a href="#cb4-49" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-50"><a href="#cb4-50" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-51"><a href="#cb4-51" tabindex="-1"></a><span class="co">#&gt; Slot &quot;res.files&quot;:</span></span>
<span id="cb4-52"><a href="#cb4-52" tabindex="-1"></a><span class="co">#&gt; $file.corrected.coverage</span></span>
<span id="cb4-53"><a href="#cb4-53" tabindex="-1"></a><span class="co">#&gt;                         file.gc.factor                file.corrected.coverage</span></span>
<span id="cb4-54"><a href="#cb4-54" tabindex="-1"></a><span class="co">#&gt; 1  output/gc/sample-0001.gc.factor.txt output/gc/sample-0001.corrected.txt.gz</span></span>
<span id="cb4-55"><a href="#cb4-55" tabindex="-1"></a><span class="co">#&gt; 2  output/gc/sample-0002.gc.factor.txt output/gc/sample-0002.corrected.txt.gz</span></span>
<span id="cb4-56"><a href="#cb4-56" tabindex="-1"></a><span class="co">#&gt; 3  output/gc/sample-0003.gc.factor.txt output/gc/sample-0003.corrected.txt.gz</span></span>
<span id="cb4-57"><a href="#cb4-57" tabindex="-1"></a><span class="co">#&gt; 4  output/gc/sample-0004.gc.factor.txt output/gc/sample-0004.corrected.txt.gz</span></span>
<span id="cb4-58"><a href="#cb4-58" tabindex="-1"></a><span class="co">#&gt; 5  output/gc/sample-0005.gc.factor.txt output/gc/sample-0005.corrected.txt.gz</span></span>
<span id="cb4-59"><a href="#cb4-59" tabindex="-1"></a><span class="co">#&gt; 6  output/gc/sample-0006.gc.factor.txt output/gc/sample-0006.corrected.txt.gz</span></span>
<span id="cb4-60"><a href="#cb4-60" tabindex="-1"></a><span class="co">#&gt; 7  output/gc/sample-0007.gc.factor.txt output/gc/sample-0007.corrected.txt.gz</span></span>
<span id="cb4-61"><a href="#cb4-61" tabindex="-1"></a><span class="co">#&gt; 8  output/gc/sample-0008.gc.factor.txt output/gc/sample-0008.corrected.txt.gz</span></span>
<span id="cb4-62"><a href="#cb4-62" tabindex="-1"></a><span class="co">#&gt; 9  output/gc/sample-0009.gc.factor.txt output/gc/sample-0009.corrected.txt.gz</span></span>
<span id="cb4-63"><a href="#cb4-63" tabindex="-1"></a><span class="co">#&gt; 10 output/gc/sample-0010.gc.factor.txt output/gc/sample-0010.corrected.txt.gz</span></span>
<span id="cb4-64"><a href="#cb4-64" tabindex="-1"></a>new.obj<span class="sc">@</span>commands <span class="ot">=</span> gc.res</span></code></pre></div>
<p>The return value of <code>correct_gc_bias</code> is an
<code>RCNA_analysis</code> object which documents the workflow
steps.</p>
<p>Alternatively, you can also choose not to estimate the GC bias and
instead use a GC factor file. We can inspect the GC factor file that we
generated from the previous step to see what the factor file should look
like.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>gc.file <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">file.path</span>(<span class="st">&quot;output&quot;</span>, <span class="st">&quot;gc&quot;</span>, <span class="st">&quot;sample-0001.gc.factor.txt&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">colnames</span>(gc.file) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Percent.GC&quot;</span>, <span class="st">&quot;Coverage.Correction.Factor&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>gc.file <span class="ot">=</span> gc.file[gc.file<span class="sc">$</span>Coverage.Correction.Factor <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">head</span>(gc.file, <span class="at">n =</span> <span class="dv">15</span>)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;    Percent.GC Coverage.Correction.Factor</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; 14       0.13                    0.14400</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 15       0.14                    0.13987</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; 16       0.15                    0.15162</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; 17       0.16                    0.16245</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; 18       0.17                    0.19365</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; 19       0.18                    0.24451</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt; 20       0.19                    0.32753</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co">#&gt; 21       0.20                    0.36505</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co">#&gt; 22       0.21                    0.50538</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="co">#&gt; 23       0.22                    0.60806</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a><span class="co">#&gt; 24       0.23                    0.63844</span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="co">#&gt; 25       0.24                    0.68623</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="co">#&gt; 26       0.25                    0.70939</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="co">#&gt; 27       0.26                    0.76110</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="co">#&gt; 28       0.27                    0.75852</span></span></code></pre></div>
<p>In the GC estimation step, a sliding window is applied to calculate
the number of G and C bases within the window. Additionally, the window
calculates the average coverage within the window normalized against
mean coverage across all bases and is constantly adjusted as the sliding
window proceeds down the coverage file. This results in a GC factor file
that has a coverage ratio associated with each GC-content bin,
corresponding to the percent of bases within a given sliding window
region that are either G or C. The size of the GC-content bins can be
adjusted using the <code>gc.step</code> parameter (default = 0.01), but
the value must tile the interval of 0 to 1.</p>
<p>So long as this column scheme is followed, a user can also provide
their own GC factor file using the <code>file.gc.factor</code> argument.
A smaller bin size may result in more finely-tuned bias correction at
the cost of increased time to calculate the factor files.</p>
</div>
<div id="estimating-normal-karyotype-range" class="section level2">
<h2>Estimating Normal Karyotype Range</h2>
<p>In the next step, we will estimate the expected “normal” karyotype
range (NKR) for each feature in the annotation file. This will be used
downstream to determine scores for features that exceed their respective
features’ NKR. This is similar to defining a confidence interval for the
coverage, however it is centered around the mode-normalized median
coverage across all samples in the analysis. Please note that the region
analyzed is strictly where the coverage file and annotation file overlap
- any regions in the coverage file that are not included within the
range of a feature in the annotation file will <strong>not</strong> be
analyzed. The ratio of coverage compared to the median is summarized in
the file passed as the <code>norm.cov.matrix</code> argument. The
normalized coverage ratio matrix will be generated if the file specified
does not exist. Additionally, the <code>nkr</code> parameter is a
quantile that defines what is considered as the “normal.” This is
considered similar to an alpha value when using bootstrapped confidence
intervals. Below is an example of how one would execute this function.
For this example, we have already specified the desired path of
<code>norm.cov.matrix</code>, as well as the value for <code>nkr</code>
(default = 0.9).</p>
<p>This step can be accomplished by using the <code>estimate_nkr</code>
function.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>nkr.res <span class="ot">&lt;-</span> <span class="fu">estimate_nkr</span>(new.obj)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt; Importing coverage matrix from previous run</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; NKR estimation succeeded!</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>nkr.res</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;RCNA_analysis&quot;</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; Slot &quot;call&quot;:</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;estimate_nkr&quot;</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; Slot &quot;params&quot;:</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; $nkrParams</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt;                         file.nkr.coverage x.norm sample.names</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; 1  output/gc/sample-0001.corrected.txt.gz  FALSE  sample-0001</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; 2  output/gc/sample-0002.corrected.txt.gz  FALSE  sample-0002</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; 3  output/gc/sample-0003.corrected.txt.gz  FALSE  sample-0003</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; 4  output/gc/sample-0004.corrected.txt.gz  FALSE  sample-0004</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt; 5  output/gc/sample-0005.corrected.txt.gz  FALSE  sample-0005</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; 6  output/gc/sample-0006.corrected.txt.gz  FALSE  sample-0006</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; 7  output/gc/sample-0007.corrected.txt.gz  FALSE  sample-0007</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt; 8  output/gc/sample-0008.corrected.txt.gz  FALSE  sample-0008</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co">#&gt; 9  output/gc/sample-0009.corrected.txt.gz  FALSE  sample-0009</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co">#&gt; 10 output/gc/sample-0010.corrected.txt.gz  FALSE  sample-0010</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="co">#&gt; $nkr</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a><span class="co">#&gt; [1] 0.9</span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a><span class="co">#&gt; $cov.ratios</span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a><span class="co">#&gt; [1] &quot;output/norm-cov-matrix.csv.gz&quot;</span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a><span class="co">#&gt; Slot &quot;res.files&quot;:</span></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a><span class="co">#&gt; $Output</span></span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ALAS2.RData&quot; &quot;SRSF2.RData&quot;</span></span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-34"><a href="#cb6-34" tabindex="-1"></a><span class="co">#&gt; $`Normalized coverage ratios`</span></span>
<span id="cb6-35"><a href="#cb6-35" tabindex="-1"></a><span class="co">#&gt; [1] &quot;output/norm-cov-matrix.csv.gz&quot;</span></span>
<span id="cb6-36"><a href="#cb6-36" tabindex="-1"></a>new.obj<span class="sc">@</span>commands[[<span class="dv">3</span>]] <span class="ot">&lt;-</span> nkr.res</span></code></pre></div>
<p>The resulting <code>.RData</code> files (found in
<code>output/nkr/</code>) contain R workspace objects which define the
normal karyotype ranges on each feature in the annotation file. There
should be one file per feature. We can load one and inspect the contents
if we so wish. Let’s also take a look at the normalized coverage matrix
too.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">file.path</span>(<span class="st">&quot;output&quot;</span>, <span class="st">&quot;nkr&quot;</span>, nkr.res<span class="sc">@</span>res.files<span class="sc">$</span>Output[<span class="dv">1</span>]))</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>raw)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt;                     sample-0001 sample-0002 sample-0003 sample-0004 sample-0005 sample-0006 sample-0007 sample-0008 sample-0009 sample-0010</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt; ALAS2_chrX_55009180  -0.4712177   0.3079704  -0.5489125  -0.4938741 -0.07685428  -0.6372187  0.06688500  0.05838189   0.8210937   0.1710256</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt; ALAS2_chrX_55009181  -0.5421495   0.3375313  -0.5301404  -0.4759917 -0.04093790  -0.5775500  0.05953157  0.02246552   0.8261610   0.2160292</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; ALAS2_chrX_55009182  -0.5991689   0.2725962  -0.5727809  -0.5267949 -0.06078825  -0.5995822  0.02852741  0.06799289   0.8366352   0.1749022</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; ALAS2_chrX_55009183  -0.5206929   0.3876275  -0.4933843  -0.4266533 -0.07423166  -0.5755403  0.04197082  0.07797758   0.8431989   0.2068355</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt; ALAS2_chrX_55009184  -0.5002292   0.3359176  -0.6075949  -0.5254258 -0.10150053  -0.5629351  0.06923969  0.10888241   0.8308551   0.2418197</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt; ALAS2_chrX_55009185  -0.5282313   0.2823252  -0.5591150  -0.5149572 -0.05550205  -0.5885885  0.09700744  0.03702967   0.8127750   0.1673071</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>norm.cov.mat <span class="ot">=</span> <span class="fu">read.csv</span>(nkr.res<span class="sc">@</span>res.files<span class="sc">$</span><span class="st">`</span><span class="at">Normalized coverage ratios</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">head</span>(norm.cov.mat)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt;               sample.0001 sample.0002 sample.0003   sample.0004 sample.0005 sample.0006 sample.0007 sample.0008 sample.0009 sample.0010</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt; _chr1_1787331  0.09619511 0.034520150  -0.2518999 -0.0409949001 -0.07862110  0.10284579  0.02632270  -0.1125486  0.10600302  -0.1263846</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; _chr1_1787332  0.07800200 0.014416556  -0.1484041 -0.0233834615 -0.14537101  0.05658341  0.01301504  -0.2154538  0.07562834  -0.1691862</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt; _chr1_1787333  0.09349493 0.054155189  -0.1638869  0.0031294940 -0.04512092  0.10563155 -0.01780169  -0.1404977  0.17946723  -0.1549712</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt; _chr1_1787334  0.02663941 0.035837190  -0.2692994  0.0210334554 -0.05555740  0.04344290 -0.03570566  -0.2147050  0.07125268  -0.1644438</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; _chr1_1787335  0.12655848 0.039913718  -0.2052132  0.0009800749 -0.10926807  0.06087082 -0.01565227  -0.1498817  0.15122988  -0.1489632</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#&gt; _chr1_1787336  0.08408666 0.006827653  -0.2353454 -0.0157945579 -0.06202690  0.09588569  0.01166820  -0.2026655  0.13805799  -0.1346327</span></span></code></pre></div>
<p>We can see that the <code>raw</code> attribute from the feature file
lines up to the rows of the normalized coverage matrix that contain that
feature. Both of these represent the normalized coverage at that
location (found in the rows) for each sample (found in the column
names). We can inspect the <code>estimate_nkr</code> results further to
find the mode-normalized median coverage and the bounds on the normal
karyotype range.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>val.pos) <span class="co"># median coverage across ALL samples</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#&gt; [1] -0.009236 -0.009236 -0.016130 -0.016130 -0.016130 -0.009236</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>uci.pos) <span class="co"># NKR upper bound</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.590188 0.606278 0.582818 0.638192 0.608133 0.574073</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>lci.pos) <span class="co"># NKR lower bound</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co">#&gt; [1] -0.597481 -0.561620 -0.599396 -0.550859 -0.587498 -0.575325</span></span></code></pre></div>
</div>
<div id="estimating-feature-score" class="section level2">
<h2>Estimating feature score</h2>
<p>In the final step of the workflow we estimate the CNA score of a
feature by calculating what percentage of the feature lands above the
upper bound calculated in the previous step. This step outputs two files
- one with <strong>all</strong> results and one with only the results
that pass a score cutoff filter specified by the
<code>score.cutoff</code> parameter(s) (default = 0.5). This parameter
is important for removing false positives, and as seen in the original
manuscript may require fine tuning to achieve an acceptable Type I error
rate.</p>
<p>This step can be accomplished by using the
<code>estimate_feature_score</code> function.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>score.res <span class="ot">&lt;-</span> <span class="fu">estimate_feature_score</span>(new.obj)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co">#&gt; Feature score estimation succeeded!</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>score.res</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#&gt; An object of class &quot;RCNA_analysis&quot;</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#&gt; Slot &quot;call&quot;:</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;estimate_feature_score&quot;</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt; Slot &quot;params&quot;:</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt; $scoreParams</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt;                       file.score.coverage sample.names</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co">#&gt; 1  output/gc/sample-0001.corrected.txt.gz  sample-0001</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt; 2  output/gc/sample-0002.corrected.txt.gz  sample-0002</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt; 3  output/gc/sample-0003.corrected.txt.gz  sample-0003</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="co">#&gt; 4  output/gc/sample-0004.corrected.txt.gz  sample-0004</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#&gt; 5  output/gc/sample-0005.corrected.txt.gz  sample-0005</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co">#&gt; 6  output/gc/sample-0006.corrected.txt.gz  sample-0006</span></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="co">#&gt; 7  output/gc/sample-0007.corrected.txt.gz  sample-0007</span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#&gt; 8  output/gc/sample-0008.corrected.txt.gz  sample-0008</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="co">#&gt; 9  output/gc/sample-0009.corrected.txt.gz  sample-0009</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a><span class="co">#&gt; 10 output/gc/sample-0010.corrected.txt.gz  sample-0010</span></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co">#&gt; $low.score.cutoff</span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a><span class="co">#&gt; [1] -0.5</span></span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a><span class="co">#&gt; $high.score.cutoff</span></span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a><span class="co">#&gt; [1] 0.5</span></span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a><span class="co">#&gt; Slot &quot;res.files&quot;:</span></span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a><span class="co">#&gt; $Output</span></span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a><span class="co">#&gt; [1] &quot;RCNA-output-scorepass.csv&quot; &quot;RCNA-output.csv&quot;</span></span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a>new.obj<span class="sc">@</span>commands[[<span class="dv">4</span>]] <span class="ot">&lt;-</span> score.res</span></code></pre></div>
<p>Let’s inspect the final results to get a better understanding of what
they mean.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>out.pass <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(new.obj<span class="sc">@</span>out.dir, <span class="st">&quot;score&quot;</span>, score.res<span class="sc">@</span>res.files<span class="sc">$</span>Output[[<span class="dv">1</span>]]), <span class="at">header =</span> T, <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>out.all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(new.obj<span class="sc">@</span>out.dir, <span class="st">&quot;score&quot;</span>, score.res<span class="sc">@</span>res.files<span class="sc">$</span>Output[[<span class="dv">2</span>]]), <span class="at">header =</span> T, <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>out.all</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt;    Sample.Name Feature Chromosome Feature.Start.Position Feature.End.Position   Score Percent.Above.NKR Percent.Below.NKR Median.Normalized.log2.Ratio Feature.Positions</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; 1  sample-0001   ALAS2       chrX               55009055             55031064 -0.0937              0.00              9.37                      -0.6073              1836</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; 2  sample-0002   ALAS2       chrX               55009055             55031064  0.0005              0.05              0.00                       0.3021              1836</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt; 3  sample-0003   ALAS2       chrX               55009055             55031064 -0.7702              0.00             77.02                      -0.6852              1836</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt; 4  sample-0004   ALAS2       chrX               55009055             55031064 -0.0223              0.00              2.23                      -0.5378              1836</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt; 5  sample-0005   ALAS2       chrX               55009055             55031064  0.0033              0.33              0.00                      -0.0006              1836</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt; 6  sample-0006   ALAS2       chrX               55009055             55031064 -0.1138              0.00             11.38                      -0.5437              1836</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt; 7  sample-0007   ALAS2       chrX               55009055             55031064  0.0000              0.00              0.00                      -0.0116              1836</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt; 8  sample-0008   ALAS2       chrX               55009055             55031064  0.0000              0.00              0.00                       0.0576              1836</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt; 9  sample-0009   ALAS2       chrX               55009055             55031064  0.9553             95.53              0.00                       0.6512              1836</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt; 10 sample-0010   ALAS2       chrX               55009055             55031064  0.0408              4.08              0.00                       0.2703              1836</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt; 11 sample-0001   SRSF2      chr17               76734115             76737374  0.0616              6.16              0.00                       0.0292               666</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt; 12 sample-0002   SRSF2      chr17               76734115             76737374  0.4715             50.45              3.30                       0.0979               666</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co">#&gt; 13 sample-0003   SRSF2      chr17               76734115             76737374  0.1291             12.91              0.00                       0.0235               666</span></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="co">#&gt; 14 sample-0004   SRSF2      chr17               76734115             76737374  0.1081             10.81              0.00                       0.0157               666</span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="co">#&gt; 15 sample-0005   SRSF2      chr17               76734115             76737374 -0.7553              1.20             76.73                      -0.2732               666</span></span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="co">#&gt; 16 sample-0006   SRSF2      chr17               76734115             76737374  0.0000              0.30              0.30                      -0.0132               666</span></span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="co">#&gt; 17 sample-0007   SRSF2      chr17               76734115             76737374 -0.0841              0.15              8.56                      -0.1064               666</span></span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a><span class="co">#&gt; 18 sample-0008   SRSF2      chr17               76734115             76737374  0.0060              1.05              0.45                      -0.1035               666</span></span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a><span class="co">#&gt; 19 sample-0009   SRSF2      chr17               76734115             76737374  0.0165              3.30              1.65                       0.0083               666</span></span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a><span class="co">#&gt; 20 sample-0010   SRSF2      chr17               76734115             76737374  0.0465             13.66              9.01                       0.0574               666</span></span></code></pre></div>
<p>A score above 0 represents a feature that was detected to have a copy
number gain, while a negative score represents a feature detected to
have a copy number loss. The <code>Percent.Above.NKR</code> and
<code>Percent.Below.NKR</code> fields represent which parts of the
feature exhibit coverage that exceed the normal karyotype range. The
last column represents the number of base pairs in the feature.</p>
</div>
<div id="run_rcna" class="section level2">
<h2>run_RCNA</h2>
<p>All of these functions can be ran in sequence by using the
<code>run_RCNA</code> function included for convenience. It has a simple
purpose - to run all three steps and return the input S4 object with the
resulting <code>RCNA_analysis</code> objects attached to the
<code>commands</code> slot. This function will always append the most
recent <code>RCNA_analysis</code> object to the end of the commands
slot, so multiple runs will have the full history if you use the same S4
object.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>res.object <span class="ot">&lt;-</span> <span class="fu">run_RCNA</span>(new.obj)</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
